### PR TITLE
chore: clarify failjobs

### PR DESCRIPTION
**Pull request by mathiasbio**
## Description

This PR tries to clarify the final analysis_status.txt which reports the status of the jobs. Currently in production it can be difficult to know what fails to pay attention to. This tries to clarify this by putting rerun jobs which later succeeded under a different heading. 

This also fixes the issue with using scontrol which only has access to job statuses within 4 days time on our cluster, instead we're using sacct now. 

#### Changed

- changed reporting of failed jobs that succeeded on restart https://github.com/Clinical-Genomics/BALSAMIC/pull/1636
- changed to using sacct instead of scontrol in job status check https://github.com/Clinical-Genomics/BALSAMIC/pull/1636

## Documentation

- [x] N/A
- [ ] Updated Balsamic documentation to reflect the changes as needed for this PR.
    - [Document Name]

### Feature Tests

- [x] Test running the updated script on a logdir with failed jobs that succeeded on rerun:

```
=== Job status check at 2025-11-17 13:20:03 ===
FAILED JOBS (no successful retry):
10717137         /full/path/CASE/logs/rule_all/10717137.log
10717187         /full/path/CASE/logs/rule_all/10717187.log

CANCELLED JOBS (no successful retry):
10717167        /full/path/CASE/logs/rule_all/10717167.log


NOTE:
Some jobs failed but succeeded on retry:
(jobid  log_path        original_state)
10714481        /full/path/CASE/logs/rule_samtools_fixmate/tumor_ACC12345/10714481.log      FAILED

```
Example below from a successful analysis:

```
=== Job status check at 2025-11-18 13:54:03 ===

SUCCESSFUL


NOTE:
Some jobs failed but succeeded on retry:
(jobid  log_path        original_state)
10714447        /home/proj/production/cancer/cases/credibleimpala/logs/rule_samtools_fixmate/tumor_ACC19308A2/10714447.log      FAILED
```

### Pipeline Integrity Tests

- **Report deliver (generation of the `.hk` file)**
    - [x] N/A
    - [ ] Verified
- **TGA T/O Workflow**
    - [x] N/A
    - [ ] Verified
- **TGA T/N Workflow**
    - [x] N/A
    - [ ] Verified
- **UMI T/O Workflow**
    - [x] N/A
    - [ ] Verified
- **UMI T/N Workflow**
    - [x] N/A
    - [ ] Verified
- **WGS T/O Workflow**
    - [x] N/A
    - [ ] Verified
- **WGS T/N Workflow**
    - [x] N/A
    - [ ] Verified
- **QC Workflow**
    - [x] N/A
    - [ ] Verified
- **PON Workflow**
    - [x] N/A
    - [ ] Verified

### Documentation

- **Atlas documentation**
    - [x] N/A
    - [ ] Updated: [Link]
- **Web portal for Clinical Genomics**
    - [x] N/A
    - [ ] Updated: [Link]

### Panel of Normal specific criteria

- [ ] The PR includes the addition of a new Panel of Normals
- [ ] The samples have been verified to adhere to the sample selection criteria on [Atlas PoN creation instructions for Balsamic](https://atlas.scilifelab.se/infrastructure/BALSAMIC/balsamic_pon/#procedure)

### User Changes

- [x] N/A
- [ ] This PR affects the output files or results.
    - [ ] User feedback is considered unnecessary because [Justification].
    - [ ] Affected users have been included in the development process and given a chance to provide feedback.

### Infrastructure Changes

- **Stored files in Housekeeper**
    - [x] N/A
    - [ ] Updated: [Link]
- **CG (CLI and delivered/uploaded files)**
    - [x] N/A
    - [ ] Updated: [Link]
- **Servers (configuration files on Hasta)**
    - [x] N/A
    - [ ] Updated: [Link]
- **Scout interface**
    - [x] N/A
    - [ ] Updated: [Link]

### Validation criteria

Validation criteria to be added to validation report PR: [LINK-TO-VALIDATION-REPORT-PR from the [validations repository](https://github.com/Clinical-Genomics/validations/)]

**Version specific criteria**

- Text here or N/A

> [!IMPORTANT]
> One of the below checkboxes for validation need to be checked 

- [ ] Added version specific validation criteria to validation report
- [ ] Changes validated in standard sections: [validation-section]
- [x] Validation criteria not necessary

## Checklist

> [!IMPORTANT]  
> Ensure that all checkboxes below are ticked before merging.

### For Developers

- **PR Description**
    - [x] Provided a comprehensive description of the PR.
    - [x] Linked relevant user stories or issues to the PR.
- **Documentation**
    - [x] Verified and updated documentation if necessary.
- **Validation criteria**
    - [x] Completed the validation criteria section of the template.
- **Tests**
    - [x] Described and tested the functionality addressed in the PR.
    - [x] Ensured integration of the new code with existing workflows.
    - [x] Confirmed that meaningful unit tests were added for the changes introduced.
    - [x] Checked that the PR has successfully passed all relevant code smells and coverage checks.
- **Review**
    - [x] Addressed and resolved all the feedback provided during the code review process.
    - [x] Obtained final approval from designated reviewers.

### For Reviewers

- **Code**
    - [ ] Code implements the intended features or fixes the reported issue.
    - [ ] Code follows the project's coding standards and style guide.
- **Documentation**
    - [ ] Pipeline changes are well-documented in the CHANGELOG and relevant documentation.
- **Validation criteria**
    - [ ] The author has completed the validation criteria section of the template
- **Tests**
    - [ ] The author provided a description of their manual testing, including consideration of edge cases and boundary
      conditions where applicable, with satisfactory results.
- **Review**
    - [ ] Confirmed that the developer has addressed all the comments during the code review.

### CODE REVIEW

File: BALSAMIC/assets/scripts/scan_finished_jobid_status.py [hunk 1]
```diff
@@ -80,31 +86,129 @@ def write_results(
...
             for jobid, log_path in cancelled:
                 out_f.write(f"{jobid}\t{log_path}\n")
             out_f.write("\n")
 
+        if resolved_failures:
+            out_f.write("\n")
+            out_f.write(
+                "NOTE:\n"
+                "Some jobs failed but succeeded on retry:\n"
+                "(jobid\tlog_path\toriginal_state)\n"
```
Comment (fevac) [inline BALSAMIC/assets/scripts/scan_finished_jobid_status.py]:
```suggestion
```
we don't have headers in the other cases, why here?
